elide-core/src/main/java/com/yahoo/elide/security/PermissionManager.java

```diff
@@ -0,0 +1,659 @@
+                throw e;
+            }
+        }
+
+        // If that succeeds, queue up our commit checks
+        if (!isEmptyCheckArray(comChecks)) {
+            commitChecks.add(() -> {
+                runPermissionChecks(comChecks, mode, resource, changeSpec, isUpdate);
+                return null;
+            });
+        }
+    }
+
+    /**
+     * Check for permissions on a class and its fields.
+     *
+     * @param resource resource
+     * @param changeSpec change spec
+     * @param annotationClass annotation class
+     * @param <A> type parameter
+     */
+    public <A extends Annotation> void checkFieldAwarePermissions(PersistentResource<?> resource,
+                                                                  ChangeSpec changeSpec,
+                                                                  Class<A> annotationClass) {
+        checkFieldAwarePermissions(resource, changeSpec, annotationClass, null);
+    }
+
+    /**
+     * Check for permissions on a specific field.
+     *
+     * @param resource resource
+     * @param changeSpec changepsec
+     * @param annotationClass annotation class
+     * @param field field to check
+     * @param <A> type parameter
+     */
+    public <A extends Annotation> void checkFieldAwarePermissions(PersistentResource<?> resource,
+                                                                  ChangeSpec changeSpec,
+                                                                  Class<A> annotationClass,
+                                                                  String field) {
+        // Select strategy
+        final Strategy strategy = (field == null)
+                                  ? new AnyFieldStrategy(resource, annotationClass, changeSpec)
+                                  : new SpecificFieldStrategy(resource, field, annotationClass, changeSpec);
+
+        // Run checks
+        strategy.executeOperationChecks();
+
+        // Queue up on success
+        if (strategy.hasCommitChecks()) {
+            commitChecks.add(() -> {
+                strategy.executeCommitChecks();
+                return null;
+            });
+        }
+    }
+
+    /**
+     * Execute commmit checks.
+     */
+    public void executeCommitChecks() {
+        commitChecks.forEach(Supplier::get);
+    }
+
+    private void runPermissionChecks(Class<? extends Check>[] checks,
+                                     CheckMode checkMode,
+                                     PersistentResource<?> resource,
+                                     ChangeSpec changeSpec,
+                                     boolean isUpdate) {
+        if (resource.getRequestScope().getSecurityMode() == SecurityMode.BYPASS_SECURITY) {
+            return;
+        }
+
+        for (Class<? extends Check> check : checks) {
+            CheckIdentifier checkId = new CheckIdentifier(resource, check);
+
+            boolean ok = (isUpdate) ? computeCheck(check, resource, changeSpec)
```

Comment:
Instead of adding a new parameter here (isUpdate) - can we just tell from whether or not the changeSpec was populated.  If the changeSpec is populated - is that not the same?